packages/client/src/components/SafeSettings/index.js

```diff
@@ -0,0 +1,22 @@
+import React, { useContext } from "react";
+import { useLocation } from "react-router-dom";
+import { Web3Context } from "contexts/Web3";
+import SafeDetailsSetting from "./components/SafeDetailsSetting";
+import SignatureThreshold from "./components/SignatureThreshold";
+import Owners from "./components/Owners";
+
+const SafeSettings = () => {
```

Comment:
Can we add treasuryAddress as input param for SafeSettings instead of fetching it from the path?

Reply:
This is to decouple the safeSettings component from the Safe page and to avoid property drill

Reply:
I've read a bit about prop drilling, "Prop drilling is basically a situation when the same data is being sent at almost every level due to requirements in the final level."
Here we would just pass this address once to this SafeSettings component, which is not that bad
Safe.js already contains fetching the address from params:
```
  const params = useParams();
  const { address, tab } = params;
```
I'm concerned about this line, and if url on staging or prod is slightly different we might encounter errors:
```
const treasuryAddress = location.pathname.split("/")[2];
```

Reply:
You are right, passing an address between those two components won't be a property drill, but it still creates dependency. The higher order of the component, the less dependent it should be with its parents, and comparing with having the key information passed down the chain,  it's actually more consistent and reliable using the pathname, the order is always gonna be what we defined in the code, refer[ this doc](https://v5.reactrouter.com/web/api/location) for the structure of location